src/main/java/com/google/cloud/genomics/utils/grpc/VariantUtils.java

```diff
@@ -60,6 +167,65 @@ public Integer apply(String string) {
       });
 
   /**
+   * Determines whether the variant is filtered according to the FILTER field,
+   * using the notation required by the VCF 4.2 standard. This means that the
+   * FILTER field of the variant must either be empty, or contain only "PASS"
+   * or ".".
+   */
+  public static final Predicate<Variant> IS_FILTERED = new Predicate<Variant>() {
+    @Override
+    public boolean apply(Variant variant) {
+      return variant.getFilterCount() != 0 && !((variant.getFilterCount() == 1)
+          && (variant.getFilter(0).equalsIgnoreCase("PASS") ||
+          variant.getFilter(0).equalsIgnoreCase(".")));
+    }
+  };
+
+  /**
+   * Determines whether a variant has one and only one alternate allele.
+   */
+  public static final Predicate<Variant> IS_BIALLELIC = Predicates.and(HAS_ALTERNATE,
+      new Predicate<Variant>() {
+        @Override
+        public boolean apply(Variant variant) {
+          return variant.getAlternateBasesList().size() == 1;
+        }
+      }
+  );
+
+  /**
+   * Determines whether a variant is a gVCF record.
+   */
+  public static final Predicate<Variant> IS_GVCF = new Predicate<Variant>() {
+    @Override
+    public boolean apply(Variant variant) {
+      return variant.getAlternateBasesCount() == 1 && variant.getAlternateBases(0).equals("<*>");
+    }
+  };
+
+  /**
+   * Determines whether a variant represents a multi-nucleotide change.
+   */
+  public static final Predicate<Variant> IS_MULTI_NUCLEOTIDE = new Predicate<Variant>() {
+    @Override
+    public boolean apply(Variant variant) {
+      int refSize = variant.getReferenceBases().length();
+      if (refSize < 1) {
+        return false;
+      } else if (refSize > 1) {
+        return true;
+      }
+      for (String alt : variant.getAlternateBasesList()) {
+        if (alt.length() != refSize) {
```

Comment:
At this point we know that `refSize == 1`.

Let's be more explicit in this check and do `if (alt.length() > refSize) {`.